src/main/python/main.py

```diff
@@ -0,0 +1,1123 @@
+        self.add_loc_info(node, imp)
+        if node.level != 0:
+            raise NotImplementedError
+        imp.setFilename(self.fname + node.module)
+
+        # make scopmanager aware of import
+        vd = VariableDeclaration()
+        self.add_loc_info(node, vd)
+        vd.setName(node.module)
+        self.scopemanager.addGlobal(vd)
+        return imp
+
+    def visit_alias(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    ### CONTROL FLOW ###
+    def visit_If(self, node):
+        debug_print(ast.dump(node))
+        stmt = IfStatement()
+        self.add_loc_info(node, stmt)
+        stmt.setCondition(self.visit(node.test))
+        stmt.setName("If")
+        if len(node.body) == 1:
+            stmt.setThenStatement(self.visit(node.body[0]))
+        else:
+            body = self.make_compound_statement(node, node.body)
+
+            stmt.setThenStatement(body)
+        if len(node.orelse) == 1:
+            stmt.setElseStatement(self.visit(node.orelse[0]))
+        else:
+            orelse = self.make_compound_statement(node, node.orelse)
+
+            stmt.setElseStatement(orelse)
+        return stmt
+
+    def make_compound_statement(self, node, node_list) -> CompoundStatement:
+        # TODO: generalize this, because this code repeats
+        body = CompoundStatement()
+        self.add_loc_info(node, body)
+
+        for b in node_list:
+            s = self.visit(b)
+
+            if s.java_name.startswith('de.fraunhofer.aisec.cpg.graph.declarations.'):
+                # wrap the statement
+                d = DeclarationStatement()
+                self.add_loc_info(node, d)
+                d.setSingleDeclaration(s)
+                body.addStatement(d)
+            else:
+                body.addStatement(s)
+
+        return body
+
+    def visit_For(self, node: ast.For):
+        debug_print(ast.dump(node))
+
+        stmt = ForEachStatement()
+        self.add_loc_info(node, stmt)
+
+        ref = self.visit(node.target)
+
+        if ref.java_name == "de.fraunhofer.aisec.cpg.graph.statements.expressions.DeclaredReferenceExpression":
+            # create a new variable
+            var = VariableDeclaration()
+            self.add_loc_info(node.target, var)
+            var.setName(ref.getName())
+
+            stmt.setVariable(var)
+        else:
+            debug_print(ref.java_name)
+            # tuple or list
+            raise NotImplementedError  # what???
+
+        stmt.setIterable(self.visit(node.iter))
+        if len(node.body) == 1:
+            stmt.setStatement(self.visit(node.body))
+        else:
+            body = self.make_compound_statement(node, node.body)
+
+            stmt.setStatement(body)
+        return stmt
+
+    def visit_While(self, node):
+        debug_print(ast.dump(node))
+        w = WhileStatement()
+        self.add_loc_info(node, w)
+        w.setCondition(self.visit(node.test))
+        body = CompoundStatement()
```

Comment:
use `make_compound_statement`

Reply:
Resolved in 5ec26c85f37b9d3fda77d70d502110e9e53196a4